Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIStorageCocoa.mm

```diff
@@ -77,8 +96,53 @@
         return;
     }
 
-    // FIXME: <https://webkit.org/b/267649> Set values for StorageAPISQLiteStore.
-    completionHandler(std::nullopt);
+    NSDictionary *data = parseJSON(dataJSON);
+    if (storageType == WebExtensionStorageType::Sync && anyItemsExceedQuota(data, webExtensionStorageAreaSyncQuotaBytesPerItem)) {
```

Comment:
can move this check to the WebProcess since it doesn't require calculating the size of any item already in storage